src/components/vm/vmMigrateDialog.jsx

```diff
@@ -149,7 +149,15 @@ export const MigrateDialog = ({ vm, connectionName, toggleModal }) => {
 
         setInProgress(true);
         return migrateToUri(connectionName, vm.id, destUri, storage, temporary)
-                .then(toggleModal, exc => {
+                .then(() => {
+                    toggleModal();
+                    if (!temporary)
+                        cockpit.location.go(["vms"]);
+                })
+                .then(() => {
+                    return getAllVms({ connectionName });
```

Comment:
Why you need this here? When the VM will get undefined we receive a event and we  update the list.

Reply:
Oh you are right, it should fire an event. I put it there because after migration finished, but the VM stayed in state, assuming I have to call getVm manually. Looks like a bug where no "undefined" event is fired when migration finished and undefined VM on source. I reported it: https://gitlab.com/libvirt/libvirt/-/issues/186